manifests/kustomize/env/platform-agnostic/mysql/mysql-deployment.yaml

```diff
@@ -15,11 +15,15 @@ spec:
       labels:
         app: mysql
     spec:
+      serviceAccountName: mysql
```

Comment:
Why is this required for non-root?

Reply:
It is not required for istio CNI configuration. For classic istio and rootless istio-init it is. 

So there are two ways for rootless istio:

1. use istio cni and get rid of the init container and any capabilities or root
2. patch the init container image to runasnonroot with NET_ADMIN and uset setpcap on the entrypoint. In this case it is good to restrict NET_ADMIN to pods with istio-init.

 So it is not necessary for the pure CNI case, but it is just good practice AFAIK to have service accounts per deployment and it helped a lot in testing. If you do not like it, I or you can remove it here and delete the new serviceaccount.yaml.

Reply:
It is not required for istio CNI configuration. For classic istio and rootless istio-init it is. 

So there are two ways for rootless istio:

1. use istio cni and get rid of the init container and any capabilities or root
2. patch the init container image to runasnonroot with NET_ADMIN and uset setpcap on the entrypoint. In this case it is good to restrict NET_ADMIN to pods with istio-init.

 So it is not necessary for the pure CNI case, but it is just good practice AFAIK to have service accounts per deployment and it helped a lot in testing. If you do not like it, i can remove it.

Reply:
I'm not against adding it, if in some scenarios it's required.

Because I did not know this before, I assume many others wouldn't too.
I think for long term maintenance, it'll be helpful adding a quick documentation with some refs for best practices required to make everything run as non-root (will the list be finite?).